aep/general/0162/aep.md.j2

```diff
@@ -1,5 +1,372 @@
+  against a previous revision.
+- An API may create data which is derived in some way from a resource at a
+  given point in time. In these cases, it may be desirable to snapshot the
+  resource for reference later.
+
+**Note:** We use the word _revision_ to refer to a historical reference for a
+particular resource, and intentionally avoid the term _version_, which refers
+to the version of an API as a whole.
+
+## Guidance
+
+APIs **may** expose a revision history for a resource. Examples of when it is
+useful include:
+
+- When it is valuable to expose older revisions of a resource via an API. This
+  can avoid the overhead of the customers having to write their own API to store
+  and enable retrieval of revisions.
+- Other resources depend on or descend from different revisions of a resource.
+- There is a need to represent the change of a resource over time.
+
+APIs implementing resources with revisions **should** model resource
+revisions as a subresource of the resource. Sometimes, the revisions
+collection can be sibling collection with the resource. it is a revision of.
+
+For example, a resource revisions which may have a longer lifespan than the
+parent resource, such as a database snapshot.
+
+{% tab proto %}
+
+```proto
+message BookRevision {
+  // The path of the book revision.
+  string path = 1;
+
+  // The snapshot of the book
+  Book resource = 2
+    [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // The timestamp that the revision was created.
+  google.protobuf.Timestamp create_time = 3
+    [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // Other revision IDs that share the same snapshot.
+  repeated string alternate_ids = 4
+    [(google.api.field_behavior) = OUTPUT_ONLY];
+}
+```
+
+- The `message` **must** be annotated as a resource (AIP-123).
+- The `message` name **must** be named `{ResourceType}Revision`.
+
+{% tab oas %}
+
+```json
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "properties": {
+        "path": {
+            "type": "string"
+        },
+        "resource": {
+            "$ref": "#/definitions/Book",
+            "readOnly": true,
+        },
+        "create_time": {
+            "type": "string",
+            "format": "date-time",
+            "readOnly": true,
+        },
+        "alternate_ids": {
+            "type": "array",
+            "items": {
+                "type": "string"
+            },
+            "readOnly": true,
+        }
+    },
+    "required": ["path"],
+}
+```
+
+{% endtabs %}
+
+- The resource revision **must** contain a field with a message type of the
+  parent resource, with a field name of `resource`.
+    - The value of `resource` **must** be the original resource
+      at the point in time the revision was created.
+- The resource revision **must** contain a `create_time` field (see [AIP-142][]).
+- The resource revision **may** contain a repeated field `aliases`, which would
```

Comment:
Sorry about that - this was a type, and "aliases" needs to be updated in the openapi. I've tried to correct all the references I can.

The idea with aliases is to align it with the existing concept in the A(IE)Ps: https://aep.dev/122#resource-id-aliases

I think regarding the tag - you're right, we are unfortunatleye repeating ourselves. It is funny - I wrote that and I think I wrote that when we didn't re-introduce "aliases" because of some internal Google feedback.

I guess the best option here is to just delete the section about tags in the history - at least now we're using a consistent term with AEPs for this (alias).

Reply:
removed section about tags.